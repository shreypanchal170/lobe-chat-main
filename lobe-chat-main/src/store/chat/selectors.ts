export { chatEnhanceSelectors } from './slices/enchance/selectors';
export { chatSelectors } from './slices/message/selectors';
export { topicSelectors } from './slices/topic/selectors';
