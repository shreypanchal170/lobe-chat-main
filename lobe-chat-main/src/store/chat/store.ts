import { PersistOptions, devtools, persist } from 'zustand/middleware';
import { shallow } from 'zustand/shallow';
import { createWithEqualityFn } from 'zustand/traditional';
import { StateCreator } from 'zustand/vanilla';

import { createHyperStorage } from '@/store/middleware/createHyperStorage';
import { isDev } from '@/utils/env';
import { isInStandaloneMode } from '@/utils/matchMedia';
import { isMobileScreen } from '@/utils/screen';

import { ChatStoreState, initialState } from './initialState';
import { ChatEnhanceAction, chatEnhance } from './slices/enchance/action';
import { ChatMessageAction, chatMessage } from './slices/message/action';
import { ShareAction, chatShare } from './slices/share/action';
import { ChatPluginAction, chatPlugin } from './slices/tool/action';
import { ChatTopicAction, chatTopic } from './slices/topic/action';

export interface ChatStoreAction
  extends ChatMessageAction,
    ChatTopicAction,
    ShareAction,
    ChatEnhanceAction,
    ChatPluginAction {}

export type ChatStore = ChatStoreAction & ChatStoreState;

//  ===============  聚合 createStoreFn ============ //

const createStore: StateCreator<ChatStore, [['zustand/devtools', never]]> = (...params) => ({
  ...initialState,

  ...chatMessage(...params),
  ...chatTopic(...params),
  ...chatShare(...params),
  ...chatEnhance(...params),
  ...chatPlugin(...params),
});

const persistOptions: PersistOptions<ChatStore> = {
  name: 'LobeChat_Chat',

  // 手动控制 Hydration ，避免 ssr 报错
  skipHydration: true,

  storage: createHyperStorage(() => {
    const isMobileAndPWA = isMobileScreen() && isInStandaloneMode();
    const selectors = [{ activeTopicId: 'topic' }];

    if (isMobileAndPWA)
      return {
        localStorage: {
          selectors,
        },
      };

    return {
      localStorage: false,
      url: {
        mode: 'hash',
        selectors,
      },
    };
  }),
  version: 0,
};

//  ===============  实装 useStore ============ //

export const useChatStore = createWithEqualityFn<ChatStore>()(
  persist(
    devtools(createStore, {
      name: 'LobeChat_Chat' + (isDev ? '_DEV' : ''),
    }),
    persistOptions,
  ),
  shallow,
);
