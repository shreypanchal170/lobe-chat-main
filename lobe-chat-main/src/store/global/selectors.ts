export * from './slices/common/selectors';
export * from './slices/settings/selectors';
