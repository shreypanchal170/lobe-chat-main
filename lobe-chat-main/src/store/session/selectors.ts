export { agentSelectors } from './slices/agent/selectors';
export { sessionExportSelectors, sessionSelectors } from './slices/session/selectors';
