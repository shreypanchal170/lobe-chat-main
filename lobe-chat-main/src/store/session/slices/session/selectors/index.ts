export * from './export';
export * from './list';
