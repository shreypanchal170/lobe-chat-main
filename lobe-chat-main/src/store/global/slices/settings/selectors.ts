import { DEFAULT_OPENAI_MODEL_LIST } from '@/const/llm';
import { DEFAULT_LANG } from '@/const/locale';
import { DEFAULT_AGENT_META } from '@/const/meta';
import {
  DEFAULT_AGENT,
  DEFAULT_AGENT_CONFIG,
  DEFAULT_SETTINGS,
  DEFAULT_TTS_CONFIG,
} from '@/const/settings';
import { Locales } from '@/locales/resources';
import { CustomModels, GlobalSettings } from '@/types/settings';
import { isOnServerSide } from '@/utils/env';
import { merge } from '@/utils/merge';

import { GlobalStore } from '../../store';

const currentSettings = (s: GlobalStore) => merge(DEFAULT_SETTINGS, s.settings);

const currentTTS = (s: GlobalStore) => merge(DEFAULT_TTS_CONFIG, s.settings.tts);

const defaultAgent = (s: GlobalStore) => merge(DEFAULT_AGENT, s.settings.defaultAgent);

const defaultAgentConfig = (s: GlobalStore) => merge(DEFAULT_AGENT_CONFIG, defaultAgent(s).config);

const defaultAgentMeta = (s: GlobalStore) => merge(DEFAULT_AGENT_META, defaultAgent(s).meta);

const openAIAPIKeySelectors = (s: GlobalStore) => s.settings.languageModel.openAI.OPENAI_API_KEY;

const openAIProxyUrlSelectors = (s: GlobalStore) => s.settings.languageModel.openAI.endpoint;

const modelListSelectors = (s: GlobalStore) => {
  let models: CustomModels = [];

  const removedModels: string[] = [];
  const modelNames = [
    ...DEFAULT_OPENAI_MODEL_LIST,
    ...(s.serverConfig.customModelName || '').split(/[,，]/).filter(Boolean),
    ...(s.settings.languageModel.openAI.customModelName || '').split(/[,，]/).filter(Boolean),
  ];

  for (const item of modelNames) {
    const disable = item.startsWith('-');
    const nameConfig = item.startsWith('+') || item.startsWith('-') ? item.slice(1) : item;
    const [name, displayName] = nameConfig.split('=');

    if (disable) {
      // Disable all models.
      if (name === 'all') {
        models = [];
      }
      removedModels.push(name);
      continue;
    }

    // Remove duplicate model entries.
    const existingIndex = models.findIndex(({ name: n }) => n === name);
    if (existingIndex !== -1) {
      models.splice(existingIndex, 1);
    }

    models.push({
      displayName: displayName || name,
      name,
    });
  }

  return models.filter((m) => !removedModels.includes(m.name));
};

export const exportSettings = (s: GlobalStore) => {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const { password: _, ...settings } = s.settings;

  return settings as GlobalSettings;
};

const currentLanguage = (s: GlobalStore) => {
  const locale = s.settings.language;

  if (locale === 'auto') {
    if (isOnServerSide) return DEFAULT_LANG;

    return navigator.language as Locales;
  }

  return locale;
};

const dalleConfig = (s: GlobalStore) => s.settings.tool?.dalle || {};
const isDalleAutoGenerating = (s: GlobalStore) => s.settings.tool?.dalle?.autoGenerate;

export const settingsSelectors = {
  currentLanguage,
  currentSettings,
  currentTTS,
  dalleConfig,
  defaultAgent,
  defaultAgentConfig,
  defaultAgentMeta,
  exportSettings,
  isDalleAutoGenerating,
  modelList: modelListSelectors,
  openAIAPI: openAIAPIKeySelectors,
  openAIProxyUrl: openAIProxyUrlSelectors,
};
