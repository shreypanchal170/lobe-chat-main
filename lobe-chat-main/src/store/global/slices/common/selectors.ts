import { GlobalStore } from '@/store/global';

import { initialState } from '../../initialState';

const sessionGroupKeys = (s: GlobalStore): string[] =>
  s.preference.sessionGroupKeys || initialState.preference.sessionGroupKeys;

const useCmdEnterToSend = (s: GlobalStore): boolean => s.preference.useCmdEnterToSend || false;

export const preferenceSelectors = {
  sessionGroupKeys,
  useCmdEnterToSend,
};
