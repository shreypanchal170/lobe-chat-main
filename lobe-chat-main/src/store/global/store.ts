import { produce } from 'immer';
import { PersistOptions, devtools, persist } from 'zustand/middleware';
import { shallow } from 'zustand/shallow';
import { createWithEqualityFn } from 'zustand/traditional';
import { StateCreator } from 'zustand/vanilla';

import { DEFAULT_AGENT, DEFAULT_LLM_CONFIG } from '@/const/settings';
import { isDev } from '@/utils/env';

import { createHyperStorage } from '../middleware/createHyperStorage';
import { type GlobalState, initialState } from './initialState';
import { type CommonAction, createCommonSlice } from './slices/common/action';
import { type SettingsAction, createSettingsSlice } from './slices/settings/action';

//  ===============  聚合 createStoreFn ============ //

export type GlobalStore = CommonAction & GlobalState & SettingsAction;

const createStore: StateCreator<GlobalStore, [['zustand/devtools', never]]> = (...parameters) => ({
  ...initialState,
  ...createCommonSlice(...parameters),
  ...createSettingsSlice(...parameters),
});

//  ===============  persist 本地缓存中间件配置 ============ //
type GlobalPersist = Pick<GlobalStore, 'preference' | 'settings'>;

const persistOptions: PersistOptions<GlobalStore, GlobalPersist> = {
  merge: (persistedState, currentState) => {
    const state = persistedState as GlobalPersist;

    return {
      ...currentState,
      ...state,
      settings: produce(state.settings, (draft) => {
        if (!draft.defaultAgent) {
          draft.defaultAgent = DEFAULT_AGENT;
        }

        // migration to new data model
        if (!draft.languageModel) {
          draft.languageModel = {
            openAI: DEFAULT_LLM_CONFIG.openAI,
          };
        }
      }),
    };
  },
  name: 'LOBE_SETTINGS',

  skipHydration: true,

  storage: createHyperStorage({
    localStorage: {
      dbName: 'LobeHub',
      selectors: ['preference', 'settings'],
    },
  }),
};

//  ===============  实装 useStore ============ //

export const useGlobalStore = createWithEqualityFn<GlobalStore>()(
  persist(
    devtools(createStore, {
      name: 'LobeChat_Global' + (isDev ? '_DEV' : ''),
    }),
    persistOptions,
  ),
  shallow,
);
