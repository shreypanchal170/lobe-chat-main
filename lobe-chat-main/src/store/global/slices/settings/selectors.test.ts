import { LanguageModel } from '@/types/llm';

import { GlobalStore } from '../../store';
import { settingsSelectors } from './selectors';

describe('settingsSelectors', () => {
  describe('currentSettings', () => {
    it('should merge DEFAULT_SETTINGS and s.settings correctly', () => {
      const s = {
        settings: {
          avatar: 'avatar.jpg',
          fontSize: 14,
          language: 'en-US',
          neutralColor: 'sand',
          password: 'password123',
          primaryColor: 'blue',
          themeMode: 'light',
          defaultAgent: {
            config: {
              systemRole: '',
              model: LanguageModel.GPT3_5,
              params: {},
              tts: {
                showAllLocaleVoice: false,
                sttLocale: 'auto',
                ttsService: 'openai',
                voice: {
                  openai: 'alloy',
                },
              },
            },
            meta: {
              avatar: 'Default Agent',
              description: 'Default agent for testing',
            },
          },
          tts: {
            openAI: {
              sttModel: 'whisper-1',
              ttsModel: 'tts-1',
            },
            sttAutoStop: true,
            sttServer: 'openai',
          },
          languageModel: {
            openAI: {
              OPENAI_API_KEY: 'openai-api-key',
              endpoint: 'https://openai-endpoint.com',
              models: ['gpt-3.5-turbo'],
            },
          },
        },
      } as unknown as GlobalStore;

      const result = settingsSelectors.currentSettings(s);

      expect(result).toMatchSnapshot();
    });
  });

  describe('CUSTOM_MODELS', () => {
    it('custom deletion, addition, and renaming of models', () => {
      const s = {
        serverConfig: {
          customModelName:
            '-all,+llama,+claude-2，-gpt-3.5-turbo,gpt-4-1106-preview=gpt-4-turbo,gpt-4-1106-preview=gpt-4-32k',
        },
        settings: {
          languageModel: {
            openAI: {},
          },
        },
      } as unknown as GlobalStore;

      const result = settingsSelectors.modelList(s);

      expect(result).toMatchSnapshot();
    });

    it('duplicate naming model', () => {
      const s = {
        serverConfig: {},
        settings: {
          languageModel: {
            openAI: {
              customModelName: 'gpt-4-1106-preview=gpt-4-turbo，gpt-4-1106-preview=gpt-4-32k',
            },
          },
        },
      } as unknown as GlobalStore;

      const result = settingsSelectors.modelList(s);

      expect(result).toMatchSnapshot();
    });

    it('should delete model', () => {
      const s = {
        serverConfig: {
          customModelName: '-gpt-4',
        },
        settings: {
          languageModel: {
            openAI: {},
          },
        },
      } as unknown as GlobalStore;

      const result = settingsSelectors.modelList(s);

      expect(result).toEqual([
        {
          displayName: 'gpt-3.5-turbo',
          name: 'gpt-3.5-turbo',
        },
        {
          displayName: 'gpt-3.5-turbo-1106',
          name: 'gpt-3.5-turbo-1106',
        },
        {
          displayName: 'gpt-3.5-turbo-16k',
          name: 'gpt-3.5-turbo-16k',
        },
        {
          displayName: 'gpt-4-32k',
          name: 'gpt-4-32k',
        },
        {
          displayName: 'gpt-4-1106-preview',
          name: 'gpt-4-1106-preview',
        },
        {
          displayName: 'gpt-4-vision-preview',
          name: 'gpt-4-vision-preview',
        },
      ]);
    });

    it('only add the model', () => {
      const s = {
        serverConfig: {},
        settings: {
          languageModel: {
            openAI: {
              customModelName: 'model1,model2,model3，model4',
            },
          },
        },
      } as unknown as GlobalStore;

      const result = settingsSelectors.modelList(s);

      expect(result).toMatchSnapshot();
    });
  });

  // describe('defaultAgent', () => {
  //   it('should merge DEFAULT_AGENT and s.settings.defaultAgent correctly', () => {
  //     const s: GlobalStore = {
  //       settings: {
  //         defaultAgent: {
  //           config: {
  //             model: 'gpt-3.5-turbo',
  //             maxTokens: 100,
  //           },
  //           meta: {
  //             name: 'Default Agent',
  //             description: 'Default agent for testing',
  //           },
  //         },
  //       },
  //     };
  //
  //     const result = settingsSelectors.defaultAgent(s);
  //
  //     expect(result).toEqual(expected);
  //   });
  // });
  //
  // describe('defaultAgentConfig', () => {
  //   it('should merge DEFAULT_AGENT_CONFIG and defaultAgent(s).config correctly', () => {
  //     const s: GlobalStore = {
  //       settings: {
  //         defaultAgent: {
  //           config: {
  //             model: 'gpt-3.5-turbo',
  //             maxTokens: 100,
  //           },
  //         },
  //       },
  //     };
  //
  //     const result = settingsSelectors.defaultAgentConfig(s);
  //
  //     const defaultAgent = settingsSelectors.defaultAgent(s);
  //     const expected = merge({}, DEFAULT_AGENT_CONFIG, defaultAgent.config);
  //
  //     expect(result).toEqual(expected);
  //   });
  // });
  //
  // describe('defaultAgentMeta', () => {
  //   it('should merge DEFAULT_AGENT_META and defaultAgent(s).meta correctly', () => {
  //     const s: GlobalStore = {
  //       settings: {
  //         defaultAgent: {
  //           meta: {
  //             name: 'Default Agent',
  //             description: 'Default agent for testing',
  //           },
  //         },
  //       },
  //     };
  //
  //     const result = settingsSelectors.defaultAgentMeta(s);
  //
  //     const defaultAgent = settingsSelectors.defaultAgent(s);
  //     const expected = merge({}, DEFAULT_AGENT_META, defaultAgent.meta);
  //
  //     expect(result).toEqual(expected);
  //   });
  // });
  //
  // describe('exportSettings', () => {
  //   it('should remove OPENAI_API_KEY and password fields from s.settings', () => {
  //     const s: GlobalStore = {
  //       settings: {
  //         OPENAI_API_KEY: 'openai-api-key',
  //         password: 'password123',
  //         avatar: 'avatar.jpg',
  //         fontSize: 14,
  //         language: 'en',
  //         neutralColor: 'white',
  //         primaryColor: 'blue',
  //         themeMode: 'light',
  //         defaultAgent: {
  //           config: {
  //             model: 'gpt-3.5-turbo',
  //             maxTokens: 100,
  //           },
  //           meta: {
  //             name: 'Default Agent',
  //             description: 'Default agent for testing',
  //           },
  //         },
  //         languageModel: {
  //           azureOpenAI: {
  //             AZURE_API_KEY: 'azure-api-key',
  //             apiVersion: 'v1',
  //             endpoint: 'https://azure-openai-endpoint.com',
  //             models: ['gpt-3.5-turbo'],
  //           },
  //           openAI: {
  //             OPENAI_API_KEY: 'openai-api-key',
  //             endpoint: 'https://openai-endpoint.com',
  //             models: ['gpt-3.5-turbo'],
  //           },
  //         },
  //       },
  //     };
  //
  //     const result = settingsSelectors.exportSettings(s);
  //
  //     const expected = {
  //       avatar: 'avatar.jpg',
  //       fontSize: 14,
  //       language: 'en',
  //       neutralColor: 'white',
  //       primaryColor: 'blue',
  //       themeMode: 'light',
  //       defaultAgent: {
  //         config: {
  //           model: 'gpt-3.5-turbo',
  //           maxTokens: 100,
  //         },
  //         meta: {
  //           name: 'Default Agent',
  //           description: 'Default agent for testing',
  //         },
  //       },
  //       languageModel: {
  //         azureOpenAI: {
  //           AZURE_API_KEY: 'azure-api-key',
  //           apiVersion: 'v1',
  //           endpoint: 'https://azure-openai-endpoint.com',
  //           models: ['gpt-3.5-turbo'],
  //         },
  //         openAI: {
  //           endpoint: 'https://openai-endpoint.com',
  //           models: ['gpt-3.5-turbo'],
  //         },
  //       },
  //     };
  //
  //     expect(result).toEqual(expected);
  //   });
  //
  //   it('should return the result as GlobalSettings type', () => {
  //     const s: GlobalStore = {
  //       settings: {
  //         avatar: 'avatar.jpg',
  //         fontSize: 14,
  //         language: 'en',
  //         neutralColor: 'white',
  //         password: 'password123',
  //         primaryColor: 'blue',
  //         themeMode: 'light',
  //         defaultAgent: {
  //           config: {
  //             model: 'gpt-3.5-turbo',
  //             maxTokens: 100,
  //           },
  //           meta: {
  //             name: 'Default Agent',
  //             description: 'Default agent for testing',
  //           },
  //         },
  //         languageModel: {
  //           azureOpenAI: {
  //             AZURE_API_KEY: 'azure-api-key',
  //             apiVersion: 'v1',
  //             endpoint: 'https://azure-openai-endpoint.com',
  //             models: ['gpt-3.5-turbo'],
  //           },
  //           openAI: {
  //             OPENAI_API_KEY: 'openai-api-key',
  //             endpoint: 'https://openai-endpoint.com',
  //             models: ['gpt-3.5-turbo'],
  //           },
  //         },
  //       },
  //     };
  //
  //     const result = settingsSelectors.exportSettings(s);
  //
  //     expect(result).toBeInstanceOf(GlobalSettings);
  //   });
  // });
});
