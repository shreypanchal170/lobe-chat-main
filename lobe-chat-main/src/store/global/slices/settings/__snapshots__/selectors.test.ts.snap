// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`settingsSelectors > CUSTOM_MODELS > custom deletion, addition, and renaming of models 1`] = `
[
  {
    "displayName": "llama",
    "name": "llama",
  },
  {
    "displayName": "claude-2",
    "name": "claude-2",
  },
  {
    "displayName": "gpt-4-32k",
    "name": "gpt-4-1106-preview",
  },
]
`;

exports[`settingsSelectors > CUSTOM_MODELS > duplicate naming model 1`] = `
[
  {
    "displayName": "gpt-3.5-turbo",
    "name": "gpt-3.5-turbo",
  },
  {
    "displayName": "gpt-3.5-turbo-1106",
    "name": "gpt-3.5-turbo-1106",
  },
  {
    "displayName": "gpt-3.5-turbo-16k",
    "name": "gpt-3.5-turbo-16k",
  },
  {
    "displayName": "gpt-4",
    "name": "gpt-4",
  },
  {
    "displayName": "gpt-4-32k",
    "name": "gpt-4-32k",
  },
  {
    "displayName": "gpt-4-vision-preview",
    "name": "gpt-4-vision-preview",
  },
  {
    "displayName": "gpt-4-32k",
    "name": "gpt-4-1106-preview",
  },
]
`;

exports[`settingsSelectors > CUSTOM_MODELS > only add the model 1`] = `
[
  {
    "displayName": "gpt-3.5-turbo",
    "name": "gpt-3.5-turbo",
  },
  {
    "displayName": "gpt-3.5-turbo-1106",
    "name": "gpt-3.5-turbo-1106",
  },
  {
    "displayName": "gpt-3.5-turbo-16k",
    "name": "gpt-3.5-turbo-16k",
  },
  {
    "displayName": "gpt-4",
    "name": "gpt-4",
  },
  {
    "displayName": "gpt-4-32k",
    "name": "gpt-4-32k",
  },
  {
    "displayName": "gpt-4-1106-preview",
    "name": "gpt-4-1106-preview",
  },
  {
    "displayName": "gpt-4-vision-preview",
    "name": "gpt-4-vision-preview",
  },
  {
    "displayName": "model1",
    "name": "model1",
  },
  {
    "displayName": "model2",
    "name": "model2",
  },
  {
    "displayName": "model3",
    "name": "model3",
  },
  {
    "displayName": "model4",
    "name": "model4",
  },
]
`;

exports[`settingsSelectors > currentSettings > should merge DEFAULT_SETTINGS and s.settings correctly 1`] = `
{
  "avatar": "avatar.jpg",
  "defaultAgent": {
    "config": {
      "autoCreateTopicThreshold": 2,
      "displayMode": "chat",
      "enableAutoCreateTopic": true,
      "historyCount": 1,
      "model": "gpt-3.5-turbo",
      "params": {
        "frequency_penalty": 0,
        "presence_penalty": 0,
        "temperature": 0.6,
        "top_p": 1,
      },
      "plugins": [],
      "systemRole": "",
      "tts": {
        "showAllLocaleVoice": false,
        "sttLocale": "auto",
        "ttsService": "openai",
        "voice": {
          "openai": "alloy",
        },
      },
    },
    "meta": {
      "avatar": "Default Agent",
      "description": "Default agent for testing",
    },
  },
  "fontSize": 14,
  "language": "en-US",
  "languageModel": {
    "openAI": {
      "OPENAI_API_KEY": "openai-api-key",
      "endpoint": "https://openai-endpoint.com",
      "models": [
        "gpt-3.5-turbo",
      ],
    },
  },
  "neutralColor": "sand",
  "password": "password123",
  "primaryColor": "blue",
  "themeMode": "light",
  "tool": {
    "dalle": {
      "autoGenerate": false,
    },
  },
  "tts": {
    "openAI": {
      "sttModel": "whisper-1",
      "ttsModel": "tts-1",
    },
    "sttAutoStop": true,
    "sttServer": "openai",
  },
}
`;
